### DISCUSSION TITLE
Blocklist Version Updates

### DISCUSSION THREAD

**Discussion by amithm7** [celzero/rethink-app#1349] [General]
Hi, had a few questions regarding how these blocklist version updates work

Info:

- I _don't_ use On-device blocklists.
- "Prompt on blocklist updates" is enabled.
- Have been using the website version, till `v055c` over the last 2 years.
- And have been using RDNS Plus mostly.

Issue:

- Every so often, I have to come here (see image attached) and check for updates manually and update the version number.
- I have never got any prompts update the blocklist [version].
- I have this issue since this button to "check for updates" exists, which is more than an year.

Questions I have:

1. What do these version updates do? Does the cloudflare / flyIO servers host multiple blocklist trie's for each weekly(?) version? If yes how long are these hosted?
2. Why don't these version update automatically? What if somebody doesn't update for more than an year? Is this a bug?
3. Even if these are supposed to manually updated for those who don't want automatic update, this button is hidden deep in the UI for something which should be more accessible for them IMO. This is at: Configure Tab -> DNS -> RethinkDNS -> Check for update -> Update. This not accessible on the "DNS" on homepage.

![image](https://github.com/celzero/rethink-app/assets/7036770/75370a8d-c72a-4bdc-afd5-b18e8f2bdc9c)

**Comment by ignoramous** [comment]
Hi there,

> check for blocklist updates:

Tracking here: https://github.com/celzero/rethink-app/issues/1353

> What do these version updates do?

For RDNS+, it refreshes blocklist *metadata* (entries count, names, groups/subgroups). This metadata rarely changes from one version to another these days (except for entries count).

For on-device blocklists, it is actually the blocklist data (blocklists crammed in trie) [along with the metadata](https://github.com/serverless-dns/blocklists/blob/11b3f1b796b5f1d1c2a6bcbb05e9cdeb07945ca0/config.json).

> Does the cloudflare / flyIO servers host multiple blocklist trie's for each weekly(?) version?

Yes. Previously, it was on S3, then moved to Lightsail Object Store. Currently, [on R2](https://github.com/serverless-dns/blocklists/blob/11b3f1b796b5f1d1c2a6bcbb05e9cdeb07945ca0/src/upload.js#L127-L128).

> If yes how long are these hosted?

Previously, we stored them forever. Recently, for 500 days.

> Why don't these version update automatically?

Some don't like the fact that things can update without their intervention (which makes sense). We do want to impl auto-updates: #564 

> What if somebody doesn't update for more than an year? 

For RDNS+, they continue using the old metadata, which is fine, as it doesn't change as much anymore.

For on-device blocklists, they're stuck on some older version. They should update.

> this button is hidden deep in the UI for something which should be more accessible for them IMO

Yeah, we need to find a proper place in the UI for this. The main DNS page is already too big with too many settings / knobs.
